REVEL.pm

```diff
@@ -0,0 +1,116 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2018] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 CONTACT
+
+ Ensembl <http://www.ensembl.org/info/about/contact/index.html>
+
+=cut
+
+=head1 NAME
+
+ REVEL
+
+=head1 SYNOPSIS
+
+ mv REVEL.pm ~/.vep/Plugins
+ ./vep -i variations.vcf --plugin REVEL,/path/to/revel/data.tsv.gz
+
+=head1 DESCRIPTION
+
+ This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
+ adds the REVEL score for missense variants to VEP output.
+
+ Please cite the REVEL publication alongside the VEP if you use this resource:
+ https://www.ncbi.nlm.nih.gov/pubmed/27666373
+
+ The tabix utility must be installed in your path to use this plugin.
+
+=cut
+package REVEL;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Utils::Sequence qw(reverse_comp);
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+sub new {
+  my $class = shift;
+
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+
+  $self->get_user_params();
+
+  return $self;
+}
+
+sub feature_types {
+  return ['Feature','Intergenic'];
+}
+
+sub get_header_info {
+  return { REVEL => 'Rare Exome Variant Ensemble Learner '};
+}
+
+sub run {
+  my ($self, $tva) = @_;
+
+  # only for missense variants
+  return {} unless grep {$_->SO_term eq 'missense_variant'} @{$tva->get_all_OverlapConsequences};
+
+  my $vf = $tva->variation_feature;
+  my $allele = $tva->variation_feature_seq;
+  my ($res) = grep {
+    $_->{alt}   eq $allele &&
+    $_->{start} eq $vf->{start} &&
+    $_->{end}   eq $vf->{end}
```

Comment:
I wonder if it is worth getting `$altaa` from the data and comparing this to `$tva->peptide` too? That way we prevent providing invalid scores for alt transcripts that might have different amino acids at this genomic location.